tidy3d/components/simulation.py

```diff
@@ -150,15 +150,20 @@ class AbstractYeeGridSimulation(AbstractSimulation, ABC):
         * `Using automatic nonuniform meshing <../../notebooks/AutoGrid.html>`_
     """
 
-    subpixel: bool = pydantic.Field(
-        True,
+    subpixel: Union[bool, SubpixelSpec] = pydantic.Field(
+        SubpixelSpec(),
         title="Subpixel Averaging",
-        description="If ``True``, uses subpixel averaging of the permittivity "
-        "based on structure definition, resulting in much higher accuracy for a given grid size.",
+        description="If ``False``, uses staircasing on structure interfaces; if ``True``, uses "
+        "default subpixel averaging methods of the permittivity based on structure definition "
+        "resulting in much higher accuracy for a given grid size. "
+        "Alternatively, user may supply a :class:`SubpixelSpec` to this field to select subpixel "
+        "averaging methods separately for dieletric, metal, and PEC materials.",
```

Comment:
Same comment as for eme sim. Side note - why is the `subpixel` field overwritten in the EME simulation currently @caseyflex ? I don't see any difference right now. I guess the thought is that in the future you may include definitions/toggles for some eme-specific subpixel in the normal direction there?

Reply:
I think it can be removed from `EMESimulation` in the same way it was removed from `Simulation`, and just left in `AbstractYeeGridSimulation`.